### ISSUE TITLE
union operator does not work on more than 2 objects

### ISSUE THREAD

**Issue by granrothge** [mcvine/SCADGen#6]
As defined for Pyre the Union operator only works on 2 objects.  However for McVine the union operator can work on multiple objects. The union operator should be updated to follow the McVine convention.   Note in the following example the third pyramid is simply ignored. 

```<!DOCTYPE geometry>

<geometry>
    <difference>
        <cylinder height="50" radius="55.0"/>
        <union>
            <translation>
                <union>
                    <rotation>
                        <pyramid edgeX="10" edgeY="10" height="55.0"/>
                        <angle>90</angle>
                        <vector>(0,1,0)</vector>
                    </rotation>
                    <rotation>
                        <pyramid edgeX="10" edgeY="10" height="55.0"/>
                        <angle>-90</angle>
                        <vector>(0,1,0)</vector>
                    </rotation>
                    <rotation>
                        <pyramid edgeX="10" edgeY="10" height="55.0"/>
                        <angle>90</angle>
                        <vector>(1,0,0)</vector>
                    </rotation>
                </union>
                <vector>(0,0,25)</vector>
            </translation>
            <cylinder height="70" radius="25.0"/>
        </union>
   </difference>
</geometry>
```

**Comment by granrothge** [comment]
I wrote a prototype (does not run) and uploaded it as a new branch.  My idea was to make an Nary class that would be similar to the Binary and Unary classes.  I tried storing the elements in a list,  but that seems to do something funny to the recursion.  @ilumsden If you want to pick up this idea or go with a different one have at it.  I just wanted to see if this would work quickly.

**Comment by ilumsden** [comment]
Thanks for letting me know about this. I was basing most of my assumptions for SCADGen on info from pyre, so I probably wouldn't have known about this if you hadn't tried it. I'll get on it right away.

**Comment by yxqd** [comment]
Thanks @ilumsden @granrothge. I forgot the subtle differences between mcvine and pyre solids. The union and intersection are good examples of N-ary operations. Actually OpenScad supports that out-of-box. For example, this example code with multiple children in a union works in OpenScad: 

```
module Torus(major, minor) {
    rotate_extrude(convexity = 10, $fn=100)
    translate([major, 0, 0])
    circle(r = minor, $fn = 100);
}

union() {
    Torus(11.25, 5);
    translate([0.0, 0.0, -5.0]) {
        cylinder(h = 10, r = 11.25, $fn=100);
    }
    cylinder(h = 8.5, r = 1, $fn=100);
    translate([-0.5, 0.0, 7.5]) {
        rotate(90.0,[0.0, 1.0, 0.0]) {
            cylinder(h = 11.25, r = 1, $fn=100);
        }
    }
}

```

**Comment by ilumsden** [comment]
@yxqd are there any other N-ary operations in McVine or is it just union and intersection? I tried looking for them, but I couldn't find them.

**Comment by yxqd** [comment]
Yes only union and intersection are N-ary.

**Comment by yxqd** [comment]
fixed in #8 . see also #7
